benchmark/utils/loggers.hpp

```diff
@@ -229,4 +229,29 @@ struct ResidualLogger : gko::log::Logger {
 };
 
 
+// Logs the number of iteration executed
+struct IterationLogger : gko::log::Logger {
+    void on_iteration_complete(const gko::LinOp *,
+                               const gko::size_type &num_iterations,
+                               const gko::LinOp *, const gko::LinOp *,
+                               const gko::LinOp *) const override
+    {
+        this->num_iters = num_iterations;
```

Comment:
When we have multiple right-hand sides: Do I understand correctly that this logs the maximum number of iterations over all rhs?